pkg/service/replay/replay.go

```diff
@@ -1408,6 +1429,130 @@ func SetTestHooks(testHooks TestHooks) {
 	HookImpl = testHooks
 }
 
+// CreateFailedTestResult creates a test result for failed test cases
+func (r *Replayer) CreateFailedTestResult(testCase *models.TestCase, testSetID string, started time.Time, errorMessage string) *models.TestResult {
```

Comment:
There is a function called `compareHTTPResp` in the same replay.go file, you can use this function to get the result and for the actual response, construct that response.